### ISSUE TITLE
After building on mac os x, dataloader complains while parsing process-conf.xml that it can't find a bad path for process-conf.xml.

### ISSUE THREAD

**Issue by tggagne** [forcedotcom/dataloader#60]
After cloning and building dataloader this AM, it complains while parsing that it can't find the path of process-conf.xml, because it's messed-up its path!  I do not have this problem with the 33.0.0 version downloaded (and copied) from installed version of the jar for the Windows download.  I only get this error using a .jar I built from the github repo.

java -cp /Users/tgagne/dataloader/dataloader-33.0.0-uber.jar:/Users/tgagne/dataloader/ojdbc6.jar -Dsalesforce.config.dir=/Users/tgagne/git/cust-sf-config/sql com.salesforce.dataloader.process.ProcessRunner process.name=importAttribute

2015-04-08 12:15:47,725 INFO  [main] process.ProcessConfig getBeanFactory (ProcessConfig.java:103) - Loading process configuration from config file: /Users/tgagne/git/cust-sf-config/sql/process-conf.xml
2015-04-08 12:15:47,872 INFO  [main] xml.XmlBeanDefinitionReader loadBeanDefinitions (XmlBeanDefinitionReader.java:315) - Loading XML bean definitions from file [/Users/tgagne/git/cust-sf-config/sql/**Users/tgagne/git/cust-sf-config/sql/process-conf.xml**]

> Notice how the conf directory is duplicated in the file[] above

2015-04-08 12:15:47,874 ERROR [main] process.ProcessConfig getProcessInstance (ProcessConfig.java:96) - Error loading process: importAttribute configuration from config file: /Users/tgagne/git/cust-sf-config/sql/process-conf.xml
org.springframework.beans.factory.BeanDefinitionStoreException: IOException parsing XML document from file [/Users/tgagne/git/cust-sf-config/sql/**Users/tgagne/git/cust-sf-config/sql/process-conf.xml**]; nested exception is java.io.FileNotFoundException: Users/tgagne/git/cust-sf-config/sql/process-conf.xml (No such file or directory)

**Comment by tggagne** [comment]
If I go back to the Dec 4 commit, eb4f47590ba909fe7f53590e8cf3e3c93ae4c619, the jar file works fine.  No errors.  I'll probably stick with this one for a bit--just to understand why upserts from databases take so long.

**Comment by diracz** [comment]
Em... this might be related to FileSystemXmlApplicationContext from spring.context. 
Pull 40: https://github.com/forcedotcom/dataloader/pull/40/files

**Comment by tggagne** [comment]
Could it be as simple as...
```
diff --git 
a/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java 
b/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java
index fac0667..78a1042 100644
--- a/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java
+++ b/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java
@@ -102,7 +102,7 @@ public class ProcessConfig {
          if (configFactory == null) {
logger.info(Messages.getFormattedString("ProcessConfig.loadingConfig", 
getConfigFileLocation()));

-            configFactory = new 
FileSystemXmlApplicationContext(getConfigFileLocation());
+            configFactory = new 
FileSystemXmlApplicationContext("file://" + getConfigFileLocation());
          }
          return configFactory;
      }
```

**Comment by tggagne** [comment]
Could it be as simple as this?  I made the change and it /seems/ to work.  If the group agrees, I'll make a pull request.
```
git diff src/main/java/com/salesforce/dataloader/process/ProcessConfig.java
diff --git a/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java b/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java
index fac0667..78a1042 100644
--- a/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java
+++ b/src/main/java/com/salesforce/dataloader/process/ProcessConfig.java
@@ -102,7 +102,7 @@ public class ProcessConfig {
         if (configFactory == null) {
             logger.info(Messages.getFormattedString("ProcessConfig.loadingConfig", getConfigFileLocation()));
             
-            configFactory = new FileSystemXmlApplicationContext(getConfigFileLocation());
+            configFactory = new FileSystemXmlApplicationContext("file://" + getConfigFileLocation());
         }
         return configFactory;
     }
```

**Comment by tggagne** [comment]
I tried creating a branch and pushing it for a subsequent pull, but I'm getting (deservedly) permission errors to the repo.

**Comment by diracz** [comment]
Would this work? This is only for windows (file://)?
This file was always like this, except the XmlBeanFactory to FileSystemXmlApplicationContext change in pull40: https://github.com/forcedotcom/dataloader/pull/40/files
Can you revert that change and see if that will work?

**Comment by tggagne** [comment]
I'm not sure what you're asking.  I haven't pushed anything.  The 
file:// works on Mac OS X.  According to the documentation, the argument 
is always considered a relative path unless preceded by "file://".

**Comment by diracz** [comment]
Okay, two points:
1. I wasn't sure if file:// would work in mac. You confirmed it worked in Mac OS X. Thanks. 
2. I suspect the root issue was caused by pull #40 . So I asked you to try if the issue can be solved by backing off pull #40 , before pushing your fix.
